libpromises/verify_vars.c

```diff
@@ -212,6 +212,11 @@ PromiseResult VerifyVarPromise(EvalContext *ctx, const Promise *pp,
             if (res.status == FNCALL_FAILURE)
             {
                 /* We do not assign variables to failed fn calls */
+                if (EvalContextGetPass(ctx) == 3) {
```

Comment:
This at least has to use a macro/constant for the last pass. Or maybe we should even have a function like `IsLastPass(ctx)`?

Reply:
> This fails in travis, as the error occurs many times in our policy:
> 
> ```
> ==25303== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
>   notice: Q: ".../cf-agent" -f /":    error: While setting variable 'tbse' in bundle 'update_def', function 'mergedata' failed - skipping
> Q: ".../cf-agent" -f /":    error: While setting variable 'tbse' in bundle 'update_def', function 'mergedata' failed - skipping
> Q: ".../cf-agent" -f /":    error: While setting variable 'control_common_ignore_missing_bundles' in bundle 'update_def', function 'ifelse' failed - skipping
> Q: ".../cf-agent" -f /":    error: While setting variable 'control_common_ignore_missing_bundles' in bundle 'update_def', function 'ifelse' failed - skipping
> ```
> 
> https://travis-ci.org/cfengine/core/jobs/483349788

I think that variable won't be set when `def.control_common_update_bundlesequence_end` is not defined, which is true unless its set from augments under `vars.control_common_update_bundlesequence_end`.

This would be good to improve, but I don't think I would make a change that started emitting output by default for this case.